dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/DefaultViewUpdatePredicate.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.scope
+
+import java.util.concurrent.TimeUnit
+
+internal class DefaultViewUpdatePredicate(
+    private val viewUpdateThreshold: Long = VIEW_UPDATE_THRESHOLD_IN_NS
+) : ViewUpdatePredicate {
+    private var lastViewUpdateTimestamp = System.nanoTime() - VIEW_UPDATE_THRESHOLD_IN_NS
+
+    override fun canUpdateView(isViewComplete: Boolean, event: RumRawEvent): Boolean {
+        val isFatalError = event is RumRawEvent.AddError && event.isFatal
+        if (isViewComplete ||
+            isFatalError ||
+            (System.nanoTime() - lastViewUpdateTimestamp > viewUpdateThreshold)
+        ) {
```

Comment:
Just a nitpick for readability

```suggestion
        val isFatalError = event is RumRawEvent.AddError && event.isFatal
        val isRequired = System.nanoTime() - lastViewUpdateTimestamp > viewUpdateThreshold
        if (isViewComplete || isFatalError || isRequired) {
```

Reply:
Yes, I wanted to add that too but then I thought that we should not compute that all the time, but you are right, it is not that much and is better for visibility